### ISSUE TITLE
Deprecated setting (TEMPLATE_CONTEXT_PROCESSORS) is required

### ISSUE THREAD

**Issue by gavinwahl** [stephenmcd/mezzanine#1364]
PageMiddleware is using the deprecated TEMPLATE_CONTEXT_PROCESSORS, which is the old and deprecated way to set context processors. The are supposed to be specified as an attribute of a template engine. I shouldn't have to use a deprecated setting to use PageMiddleware.

**Comment by jerivas** [comment]
I think it's deprecated in Django 1.8, but not in 1.7, which Mezzanine
supports too. I guess settings will be fully converted to the new format
when we drop Django 1.7 and support 1.8 and 1.9 only.

**Comment by aibon** [comment]
This also breaks the check set_dynamic_settings() in utils/conf.py where TEMPLATE_CONTEXT_PROCESSORS is checked. I'm not even sure if the context processors will work in a django 1.8 installation, supposed that they should be defined in a Template engine settins, seems not as far as i checked.

**Comment by koobs** [comment]
I'm hitting this too, is there a workaround?

**Comment by jjlorenzo** [comment]
This is my workaround:
```python
TEMPLATES = [
  {
    'BACKEND': 'django.template.backends.django.DjangoTemplates',
    'DIRS': (os.path.join(PROJECT_ROOT, 'templates'),),
    'APP_DIRS': True,
    'OPTIONS': {
      'context_processors': (
        'django.contrib.auth.context_processors.auth',
        'django.contrib.messages.context_processors.messages',
        'django.core.context_processors.debug',
        'django.core.context_processors.i18n',
        'django.core.context_processors.static',
        'django.core.context_processors.media',
        'django.core.context_processors.request',
        'django.core.context_processors.tz',
        'mezzanine.conf.context_processors.settings',
        'mezzanine.pages.context_processors.page',
      ),
    },
  },
]

# mezzanine hacks to works with django LTS (1.8.x)
TEMPLATE_CONTEXT_PROCESSORS = TEMPLATES[0]['OPTIONS']['context_processors']
TEMPLATE_DIRS = TEMPLATES[0]['DIRS']

SILENCED_SYSTEM_CHECKS = (
  '1_8.W001'
)
```

**Comment by AlexHill** [comment]
This will be addressed in #1481 where we'll drop 1.7 support in line with Django's support policy, and migrate everything to the `TEMPLATES` settings style.
